avocado/utils/asset.py

```diff
@@ -117,9 +125,15 @@ def fetch(self):
                         path = urlobj.path
                     log.debug('Looking for file on %s.' % path)
                     if self._check_file(path):
-                        os.symlink(path, self.asset_file)
+                        try:
+                            os.symlink(path, self.asset_file)
+                        except OSError as e:
+                            if e.errno == errno.EEXIST:
+                                os.remove(self.asset_file)
+                                os.symlink(path, self.asset_file)
```

Comment:
We already check if the link exists in `_check_file()`. We only get here if the link exists but it's *expired*. Seems like a very unusual scenario (set expire time for an asset present in a `file://` location). So, instead of run an `if os.path.exists()` on every single execution, I'd prefer to only care about that when the unusual scenario happens. That's when the except comes handy. So, if you can live with this, I'd prefer to keep it.